src/modules/multi-chain-share/DestinationMinter.sol

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.21;
+
+import { SafeTransferLib } from "@solmate/utils/SafeTransferLib.sol";
+import { ERC20 } from "@solmate/tokens/ERC20.sol";
+import { CCIPReceiver } from "@ccip/contracts/src/v0.8/ccip/applications/CCIPReceiver.sol";
+import { Client } from "@ccip/contracts/src/v0.8/ccip/libraries/Client.sol";
+import { IRouterClient } from "@ccip/contracts/src/v0.8/ccip/interfaces/IRouterClient.sol";
+
+/**
+ * @title DestinationMinter
+ * @notice Receives CCIP messages from SourceLocker, to mint ERC20 shares that
+ *         represent ERC4626 shares locked on source chain.
+ * @author crispymangoes
+ */
+contract DestinationMinter is ERC20, CCIPReceiver {
+    using SafeTransferLib for ERC20;
+
+    //============================== ERRORS ===============================
+
+    error DestinationMinter___SourceChainNotAllowlisted(uint64 sourceChainSelector);
+    error DestinationMinter___SenderNotAllowlisted(address sender);
+    error DestinationMinter___InvalidTo();
+    error DestinationMinter___FeeTooHigh();
```

Comment:
So this error only bubbles up if the user specified `maxLinkToPay` is less than the fees it would cost to send the bridge TX. So we could probably add in 2 args, the user specified max, and the actual fees being charged. However I dont think etherscan knows how to process custom errors, so this would only help people interacting with our contracts via some dev env that tells them.